### PR TITLE
Sokol integration and more

### PR DESCRIPTION
**Pull request by luzzif**
Sokol integration and full compatibility with those chains that don't have Ether as the native currency (i.e. Sokol with SPOA and xDAI with xDAI).

### CODE REVIEW

File: src/components/CurrencyLogo/index.tsx [hunk 1]
```diff
@@ -33,24 +35,23 @@ export default function CurrencyLogo({
...
+  const { chainId } = useWeb3React()
+  const nativeCurrencyLogo = NATIVE_CURRENCY_LOGO[(chainId as ChainId) || ChainId.MAINNET]
+
   const srcs: string[] = useMemo(() => {
-    if (currency === ETHER) return []
+    if (currency && Currency.isNative(currency) && !!nativeCurrencyLogo) return [nativeCurrencyLogo]
 
     if (currency instanceof Token) {
-      if (currency.name === 'DXdao') {
+      if (currency?.name === 'DXdao') {
```
Comment (jpkcambridge) [inline src/components/CurrencyLogo/index.tsx:45]: There's a question mark?  (this is a question :) )

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -237,17 +238,16 @@ const StyledExternalLink = styled(ExternalLink).attrs({
...
+  [ChainId.SOKOL]: 'Sokol'
 }
 
 function Header({ history }: { history: any }) {
   const { account, chainId } = useActiveWeb3React()
   const { t } = useTranslation()
 
-  const userEthBalance = useETHBalances(account ? [account] : [])?.[account ?? '']
+  const nativeCurrency = useNativeCurrency()
+  const userNativeCurrencyBalance = useNativeCurrencyBalances(account ? [account] : [])?.[account ?? '']
```
Comment (jpkcambridge) [inline src/components/Header/index.tsx:250]: So many question marks :)

File: src/connectors/index.ts [hunk 3]
```diff
@@ -2,19 +2,19 @@ import { InjectedConnector } from '@web3-react/injected-connector'
...
-    1: `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`,
-    4: `https://rinkeby.infura.io/v3/${INFURA_PROJECT_ID}`
+    1: `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`
   },
   defaultChainId: 1
 })
 
 export const injected = new InjectedConnector({
-  supportedChainIds: [1, 4, 79377087078960]
+  supportedChainIds: [ChainId.MAINNET, ChainId.RINKEBY, ChainId.ARBITRUM_TESTNET_V3, ChainId.SOKOL]
```
Comment (jpkcambridge) [inline src/connectors/index.ts:17]: No xdai?

File: src/constants/abis/wspoa.json [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  },
+  {
+    "type": "function",
+    "stateMutability": "nonpayable",
+    "payable": false,
+    "outputs": [{ "type": "bool", "name": "" }],
+    "name": "approve",
+    "inputs": [
+      { "type": "address", "name": "guy" },
+      { "type": "uint256", "name": "wad" }
```
Comment (jpkcambridge) [inline src/constants/abis/wspoa.json:19]: What is "guy" and "wad"?

File: src/state/swap/hooks.ts [hunk 5]
```diff
@@ -74,7 +76,9 @@ export function tryParseAmount(value?: string, currency?: Currency): CurrencyAmo
     if (typedValueParsed !== '0') {
       return currency instanceof Token
         ? new TokenAmount(currency, JSBI.BigInt(typedValueParsed))
-        : CurrencyAmount.ether(JSBI.BigInt(typedValueParsed))
+        : chainId
+        ? CurrencyAmount.nativeCurrency(JSBI.BigInt(typedValueParsed), chainId)
```
Comment (jpkcambridge) [inline src/state/swap/hooks.ts:80]: Is this two ternaries?

**Review by jpkcambridge — APPROVED** [review]

File: src/components/CurrencyLogo/index.tsx (hunk 1 above)
Comment (luzzif) [reply src/components/CurrencyLogo/index.tsx:45]: Ooops, my bad. That's indeed redundant, will remove it

File: src/connectors/index.ts (hunk 3 above)
Comment (luzzif) [reply src/connectors/index.ts:17]: Will be added as soon as I deploy the contracts there and start the whole integration (so I'd say pretty soon!)

File: src/constants/abis/wspoa.json (hunk 4 above)
Comment (luzzif) [reply src/constants/abis/wspoa.json:19]: Pretty sure `guy` is just the address that you are allowing to spend `wad` amount of your `WSPOA`. A `wad` is just an integer representation of a number with 18 decimals (so wei with the added info that it refers to a 18-decimal number).

File: src/state/swap/hooks.ts (hunk 5 above)
Comment (luzzif) [reply src/state/swap/hooks.ts:80]: Yes... it's not the best for clarity. I will rework it.

File: src/components/CurrencyLogo/index.tsx (hunk 1 above)
Comment (luzzif) [reply src/components/CurrencyLogo/index.tsx]: bd405cd2253c9651af25625903ff4546a35b94ea

File: src/state/swap/hooks.ts (hunk 5 above)
Comment (luzzif) [reply src/state/swap/hooks.ts]: 13f1e0bece83146d489d1c53ae393f4de03d27ad

File: src/components/Header/index.tsx (hunk 2 above)
Comment (luzzif) [reply src/components/Header/index.tsx]: Simplified a little bit (split into 2 instructions): 1ba5023bb7a6a550e95eef08d9fe008c6b0df2f1

**Review by nicoelzer — APPROVED** [review]

**Pull request merged**